src/index.ts

```diff
@@ -4,10 +4,18 @@
  *************************************************************************************************/
 
 import * as ghCore from "@actions/core";
+import * as path from "path";
+import * as os from "os";
 import { Inputs, Outputs } from "./generated/inputs-outputs";
 import Deploy from "./deploy";
 import * as utils from "./utils";
 
+// boolean value to check if pull secret is created or not
+let isPullSecretCreated = false;
```

Comment:
can't we scope these variables to the run function and cut down on the amount of global state?

Reply:
I don't think we can cut down these variables from global state as these are required after run finishes it's execution. i.e. to delete the created secret
```
 .then(async () => {
        ghCore.info("Success.");
        if (isPullSecretCreated) {
            await Deploy.deletePullSecretWithLabel(pullSecretName, namespaceArg);
        }
    })
```

Reply:
you could return the pull secret from run